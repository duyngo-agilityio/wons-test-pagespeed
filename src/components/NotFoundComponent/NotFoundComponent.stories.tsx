import type { Meta, StoryObj } from '@storybook/react';

// Components
import NotFoundComponent from '.';

const meta: Meta<typeof NotFoundComponent> = {
  title: 'Components/NotFoundComponent',
  component: NotFoundComponent,
  parameters: {
    layout: 'centered',
    controls: { expanded: true },
  },
};

export default meta;
type Story = StoryObj<typeof NotFoundComponent>;

export const Default: Story = {
  args: {},
};
