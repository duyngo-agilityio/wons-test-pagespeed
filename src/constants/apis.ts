export const API_BASE_URL = process.env.NEXT_API_BASE_URL || '';

export const API_PATH = {
  SIGN_UP: '/auth/local/register',
  USERS: '/users',
};
