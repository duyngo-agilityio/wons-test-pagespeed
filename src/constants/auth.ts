// Types
import { TRole } from '@/types';

export const AUTH_METHODS = {
  CREDENTIALS: 'credentials',
};

export const ERROR_TYPES = {
  CREDENTIALS_SIGN_IN: 'CallbackRouteError',
};

export const ROLES: TRole[] = [
  {
    id: 3,
    name: 'Admin',
  },
  {
    id: 4,
    name: 'User',
  },
];
