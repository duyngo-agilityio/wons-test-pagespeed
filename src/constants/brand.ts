export const BRANDS = [
  { key: 'apple', label: 'Apple' },
  { key: 'samsung', label: 'Samsung' },
  { key: 'huawei', label: 'Huawei' },
  { key: 'xioami', label: 'Xioami' },
  { key: 'oppo', label: 'Oppo' },
  { key: 'google', label: 'Google' },
];
