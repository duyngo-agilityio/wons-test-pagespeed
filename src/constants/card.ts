export enum LevelCard {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
}
