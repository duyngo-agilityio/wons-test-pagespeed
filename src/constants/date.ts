export const DEFAULT_RANGE_VALUE_PICKER = 7;
