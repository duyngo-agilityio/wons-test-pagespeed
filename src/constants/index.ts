export * from './apis';
export * from './auth';
export * from './date';
export * from './fonts';
export * from './images';
export * from './messages';
export * from './params';
export * from './regex';
export * from './routes';
export * from './sidebar';
export * from './themes';
export * from './typography';
export * from './messages';
export * from './regex';
export * from './apis';
export * from './auth';
export * from './invoices';
export * from './select';
export * from './pagination';
