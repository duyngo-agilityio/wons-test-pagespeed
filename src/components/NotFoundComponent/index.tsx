import Link from 'next/link';

// Constants
import { ROUTES } from '@/constants';

// Components
import { Button, Heading, Text } from '../common';

const NotFoundComponent = () => (
  <div className="text-center h-full flex justify-center items-center flex-col">
    <Heading title="Not Found" className="text-6xl" />
    <Text text="Could not find requested resource" className="mt-4" />
    <Link
      href={ROUTES.DASHBOARD}
      style={{ textDecoration: 'underline', textAlign: 'center' }}
    >
      <Button color="primary" className="mt-4">
        Return Home
      </Button>
    </Link>
  </div>
);

export default NotFoundComponent;
