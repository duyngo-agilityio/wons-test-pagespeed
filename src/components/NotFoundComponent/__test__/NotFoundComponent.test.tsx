import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

// Constants
import { ROUTES } from '@/constants';

// Components
import NotFoundComponent from '../index';

describe('NotFoundComponent', () => {
  it('renders Not Found title and error message', () => {
    render(<NotFoundComponent />);

    expect(screen.getByText('Not Found')).toBeInTheDocument();
    expect(
      screen.getByText('Could not find requested resource'),
    ).toBeInTheDocument();
  });

  it('renders link to return home with correct route', () => {
    render(<NotFoundComponent />);

    const link = screen.getByRole('link', { name: /return home/i });
    expect(link).toHaveAttribute('href', ROUTES.DASHBOARD);
  });

  it('renders the button with correct text', () => {
    render(<NotFoundComponent />);

    const button = screen.getByText('Return Home');
    expect(button).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<NotFoundComponent />);
    expect(container).toMatchSnapshot();
  });
});
