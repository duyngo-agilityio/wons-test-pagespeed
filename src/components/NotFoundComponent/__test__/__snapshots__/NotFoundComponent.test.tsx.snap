// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundComponent matches snapshot 1`] = `
<div>
  <div
    class="text-center h-full flex justify-center items-center flex-col"
  >
    <h2
      aria-level="2"
      class="font-bold font-dm-sans text-blue-800 dark:text-white text-5xl text-6xl"
    >
      Not Found
    </h2>
    <p
      class="font-dm-sans text-blue-800 dark:text-white text-md mt-4"
    >
      Could not find requested resource
    </p>
    <a
      href="/"
      style="text-decoration: underline; text-align: center;"
    >
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 gap-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover bg-blue-500 border-transparent dark:bg-purple-600 text-white hover:bg-blue-100 px-5 pb-2.5 pt-[11px] text-md rounded-[10px] border-[1px] h-auto min-w-max mt-4"
        type="button"
      >
        <div
          class="flex items-center"
        >
          Return Home
        </div>
      </button>
    </a>
  </div>
</div>
`;
